src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIPacket.cs

```diff
@@ -249,7 +237,7 @@ public void ReadFromStreamAsync(Stream stream, SNIAsyncCallback callback, bool i
                 options |= TaskContinuationOptions.LongRunning;
             }
 
-            stream.ReadAsync(_data, 0, _capacity).ContinueWith(t =>
+            stream.ReadAsync(_data, 0, _data.Length).ContinueWith(t =>
```

Comment:
Not related to your change, but looking at this code...

I assume this ReadFromStreamAsync method is on a hot path?  As it stands, it's always going to queue a continuation, even if the stream.ReadAsync completes synchronously, which in general is a common thing (I don't know about this specific code and usage patterns), and it's always going to incur a closure, so every time ReadFromStreamAsync is called, it looks like there's at least four allocations of additional overhead happening here just from the way the method is constructed.  It'd be great to look at rewriting the async code paths through the library to use async/await, as I expect that'll help significantly with maintainability and also help with perf, but if that's too much work for the short-term, you might look at rewriting just this method to use async/await (it could be an async void method).

Reply:
Thanks @stephentoub 
@geleems Lets take this as a fix to a new issue that I have opened here https://github.com/dotnet/corefx/issues/27222 
I will be worth considering all I/O calls in SqlClient managed SNI and change them  to use async/await instead of creating continuation tasks.